src/reader.cpp

```diff
@@ -3511,15 +3513,45 @@ void reader_data_t::handle_readline_command(readline_cmd_t c, readline_loop_stat
             break;
         }
         case rl::backward_word:
-        case rl::backward_bigword: {
+        case rl::backward_bigword:
+        case rl::prevd_or_backward_word: {
+            if (c == rl::prevd_or_backward_word && command_line.empty()) {
+              // Execute the prevd function. We capture and reprint output because prevd signals
+              // that you are already at the beginning of the directory history with a bell.
+              // TODO: This is duplicated in nextd_or_forward_word. A "exec and print to outputter"
+              // function might be helpful.
+              outputter_t &outp = outputter_t::stdoutput();
+              wcstring_list_t outputs;
+              (void)exec_subshell(L"prevd", parser(), outputs, false /* ignore exit status */);
```

Comment:
I'm pretty sure we can just write

	(void)parser().eval(L"prevd", io_chain_t{});

that seems to inherit the output streams

Reply:
Thanks, that is indeed better. (We do need to be careful about restoring the status.)